1036/en.md

```diff
@@ -0,0 +1,185 @@
+
+Let's take `test case 2` as an example.
+
+`uraniumMatrix` :
+|   i\j| 0  |1   |2   |3   |
+|---|---|---|---|---|
+|   __0__|0   |0   |0   |0   |
+|   __1__|  0 |5   |15   |49   |
+|   __2__ |   0|  0 |0   |0   |
+|   __Total Uranium Taken From The Column </br> (M<sub>0</sub>,<sub>j</sub> + M<sub>1</sub>,<sub>j</sub> + ... + M<sub>n</sub>,<sub>j</sub>)__ | __0__ |  __5__ |__15__   |__49__   |
+
+`radiumMatrix` :
+|   i\j| 0  |1   |2   |3   |Total Radium Taken From The Row </br> (M<sub>i</sub>,<sub>0</sub> + M<sub>i</sub>,<sub>1</sub> + ... + M<sub>i</sub>,<sub>n</sub>) |
+|---|---|---|---|---|---|
+|   __0__|0   |0   |0   |0   | __0__|
+|   __1__|  0 |0   |0   |0   | __0__|
+|   __2__|   50|  0 |0   |0   | __50__|
+
+Now for the block-wise decision-making of which type of conveyor it should be part of, we have a third 2D array that will compare both the uranium and radium matrices. Since `uraniumMatrix` was left to right cumulated, we need to add up from the upper block of the `memoizationMatrix` of the current block. From which resource the best cumulation was taken for that upper block is not relevant as we had cumulated the refinery matrices in the opposite direction of the refineries, thus cross-paths have been handled. To intuitively understand this, look at the last row of the `uraniumMatrix`, if we sum up neighboring blocks of the same column of the `uraniumMatrix`, we are calculating how much uranium was taken up to that particular column. As we are to choose the best, we are taking the updated value from `memoizationMatrix`. In other words, the `memoizationMatrix` works as the updated value holder for summing up. Similarly, it’s the immediate left block for adding to `radiumMatrix` as it memoized from up to down. Basically, `memoizationMatrix[i][j]` is the modified value for `uraniumMatrix[i][j]` and `radiumMatrix[i][j]`. And we are progressing as if we are comparing just 2 matrices. This is how we are doing the bottom-up approach. And the last block of the `memoizationMatrix` will be the answer.
+
+`memoizationMatrix` :
+|   i\j| 0  |1   |2   |3   |
+|---|---|---|---|---|
+|   __0__|0   |0   |0   |0   |
+|   __1__|0 |5   |15   |49   |
+|   __2__|   0|  50 |50   |__50__   |
+
+The above implementation is `accepted`.
+
+## Time Complexity
+
+2D Array -> mutation : O(1)
+
+2D Array -> look up : O(1)
+
+Whole 2D Array -> iteration : O(row x column)
+
+Overall Complexity : O(3 x row x column) = O(row x column)
```

Comment:
This is all implementation details, i.e. how you implemented your solution. 
This is not so much helpful to anybody else. 
Instead, write up what the solution is. Since this is a DP problem, you would probably state how and what the sub-problems are, what the states will be for a sub-problem, the base cases, the transitions etc.
You don't need to add why you should pad an array or which cells of a 2-D array should you look up.

Reply:
I will review your code after I understand the solution.

Reply:
Resolved by sections:
- What are the 2 subproblems?
- Why does direction matter in the case of an individual refinery accumulation to avoid cross paths? 
- Explained transition and states: cumulating by taking from `radiumMatrix` or `uraniumMatrix`.
- Why `memoizedMatrix[i][j]` value obtained is irrelevant of which matrix it was obtained from?
- Why are updated values from `memoizedMatrix` cross-path free?
- Why original matrices can't be over-written?